### ISSUE TITLE
numba is a dependency?

### ISSUE THREAD

**Issue by jasmainak** [alphacsc/alphacsc#11]
I got this:

```py
In [1]: %run examples/multicsc/plot_sample_evoked_response.py
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
~/Desktop/projects/github_repos/alphacsc/alphacsc/utils/compat.py in <module>()
      4 try:
----> 5     import numba
      6     from numba import jit

ModuleNotFoundError: No module named 'numba'

During handling of the above exception, another exception occurred:

AttributeError                            Traceback (most recent call last)
~/Desktop/projects/github_repos/alphacsc/examples/multicsc/plot_sample_evoked_response.py in <module>()
     41 # Next, we define the parameters for multivariate CSC
     42 
---> 43 from alphacsc import GreedyCDL
     44 cdl = GreedyCDL(
     45     # Shape of the dictionary

~/Desktop/projects/github_repos/alphacsc/alphacsc/__init__.py in <module>()
----> 1 from .update_d import update_d_block
      2 from .learn_d_z import learn_d_z, objective
      3 from .learn_d_z_mcem import learn_d_z_weighted
      4 from .learn_d_z_multi import learn_d_z_multi
      5 from .utils import construct_X, check_random_state

~/Desktop/projects/github_repos/alphacsc/alphacsc/update_d.py in <module>()
      7 from scipy import linalg, optimize
      8 
----> 9 from .utils import construct_X, check_consistent_shape
     10 
     11 

~/Desktop/projects/github_repos/alphacsc/alphacsc/utils/__init__.py in <module>()
      1 # flake8: noqa F401
      2 from .dictionary import get_D, get_uv
----> 3 from .convolution import construct_X, construct_X_multi, _choose_convolve
      4 from .validation import check_random_state, check_consistent_shape
      5 from .profile_this import profile_this

~/Desktop/projects/github_repos/alphacsc/alphacsc/utils/convolution.py in <module>()
     10 
     11 from .. import cython_code
---> 12 from .compat import numba, jit
     13 from .lil import get_z_shape, is_lil
     14 

~/Desktop/projects/github_repos/alphacsc/alphacsc/utils/compat.py in <module>()
     16 
     17     numba = object()
---> 18     numba.int64 = dummy_numba_type()
     19     numba.float64 = dummy_numba_type()
     20 

AttributeError: 'object' object has no attribute 'int64'
```
Should the readme be updated or `numba` is not meant to be a dependency?

**Comment by tomMoral** [comment]
`numba` is a requirement and it is listed in the `requirements.txt` file. It should automatically be installed when using `pip`. Did you have an issue with it?

We do not list the dependency in the `README` but we could if you think it could clarify. We only list `numpy` and `cython` as the `setup.py` depends on them.

**Comment by agramfort** [comment]
why do you have a compat.py file that seems to try to mock numba then?

**Comment by TomDLT** [comment]
We should remove it.

**Comment by TomDLT** [comment]
solved in 7b4ed166d9eaa02878e9bb450ffe044bcb08ec24.
